From 8c53aa3933cf51abd326ba5bc5811d04cf19488a Mon Sep 17 00:00:00 2001
From: Patrick Uiterwijk <puiterwijk@redhat.com>
Date: Jun 03 2017 23:17:44 +0000
Subject: Make sure to fix encoding all RPM Headers


The fix in #403 left some RPM Headers unfixed, most notably changelog* (because those are lists).
This patch makes use of fixEncodingRecurse to make sure all heades get encoded correctly.

Signed-off-by: Patrick Uiterwijk <puiterwijk@redhat.com>

---

diff --git a/hub/kojihub.py b/hub/kojihub.py
index 05dfb27..6257f20 100644
--- a/hub/kojihub.py
+++ b/hub/kojihub.py
@@ -9878,10 +9878,7 @@ class RootExports(object):
             raise koji.GenericError('either rpmID or taskID and filepath must be specified')
 
         headers = koji.get_header_fields(rpm_path, headers)
-        for key, value in headers.items():
-            if isinstance(value, basestring):
-                headers[key] = koji.fixEncoding(value, remove_nonprintable=True)
-        return headers
+        return koji.fixEncodingRecurse(headers, remove_nonprintable=True)
 
     queryRPMSigs = staticmethod(query_rpm_sigs)
 

